resolver/resolver.cc

```diff
@@ -527,7 +528,10 @@ class ResolveConstantsWalk {
             }
         } else {
             ENFORCE(resolved.data(ctx)->isClassOrModule());
-            job.klass.data(ctx)->addMixin(resolved);
+            // Don't mutate mixins during incremental resolver.
+            if (!isIncrementalResolver) {
```

Comment:
can we make it idempotent by design rather than hardcoding it? e.g. test if it's already there.

Reply:
Sure, but that'll require a vector crawl on every `addMixin`.

Reply:
I can try that out and measure the perf.

Reply:
> Sure, but that'll require a vector crawl on every addMixin.
> I can try that out and measure the perf.

Alternatively, let's ENFORCE it so that we know it for sure.

Reply:
On second thought, I don't know if I can do this. Ancestor order matters. Testing if it is there isn't sufficient.

Reply:
e.g., the following case would be broken:

```
class Foo
  include Bar
  include Baz
  include Bar
end
```

In this scenario, `Bar`'s methods shadow `Baz` because it is included after `Baz`.

Reply:
Actually, I think we can do this. Symbols have a bit set when their mixins are linearized. If that bit is set, I can opt into the membership check. If the mixin isn't part of the vector, add it.

I wish we could just make addMixin a no-op when the symbol is linearized, but unfortunately we _do_ have to support mixing in classes to linearized symbols. This is how we support mixins on classes/modules defined in payload.

Reply:
I have this implemented and am currently debugging how/why we are still adding new mixins to classes during incremental resolver -- mixins which should have been added the first time through.

Reply:
Done.